### PR TITLE
adding backend changes for EMR and fixing some linting issues

### PR DESCRIPTION
**Pull request by ahmadnazeri**
# Description

### Select type(s) of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

# Checklist:

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have fixed any merge conflicts

### CODE REVIEW

File: provider/spark.go [hunk 1]
```diff
@@ -37,30 +37,45 @@ type SparkExecutorType string
 
 const (
 	EMR        SparkExecutorType = "EMR"
-	Databricks                   = "DATABRICKS"
+	Databricks SparkExecutorType = "DATABRICKS"
```
Comment (sdreyer) [inline provider/spark.go:40]: Dont need to do this, the type is inferred from the previously declared constant

File: provider/spark.go [hunk 2]
```diff
@@ -37,30 +37,45 @@ type SparkExecutorType string
...
 )
 
 type JobType string
 
 const (
 	Materialize       JobType = "Materialization"
-	Transform                 = "Transformation"
-	CreateTrainingSet         = "Training Set"
+	Transform         JobType = "Transformation"
```
Comment (sdreyer) [inline provider/spark.go:47]: JobType not needed here

File: provider/spark.go [hunk 3]
```diff
@@ -37,30 +37,45 @@ type SparkExecutorType string
...
+}
 
 type SparkConfig struct {
 	ExecutorType   SparkExecutorType
-	ExecutorConfig DatabricksConfig
+	ExecutorConfig SparkExecutorConfig
 	StoreType      FileStoreType
-	StoreConfig    AzureFileStoreConfig
-} //TODO, change these back to type agnostic after databricks tests
+	StoreConfig    SparkFileStoreConfig
```
Comment (sdreyer) [inline provider/spark.go:77]: Does this need to be SparkFileStoreConfig or can it just be FilestoreConfig?

File: provider/spark.go [hunk 4]
```diff
@@ -105,13 +119,17 @@ func (e *EMRConfig) Serialize() []byte {
...
+}
+
 type DatabricksResultState string
 
 const (
 	Success   DatabricksResultState = "SUCCESS"
-	Failed                          = "FAILED"
-	Timeout                         = "TIMEOUT"
-	Cancelled                       = "CANCELLED"
+	Failed    DatabricksResultState = "FAILED"
```
Comment (sdreyer) [inline provider/spark.go:130]: DatabricksResultState not needed

File: provider/spark.go [hunk 5]
```diff
@@ -454,18 +482,16 @@ func sparkOfflineStoreFactory(config SerializedConfig) (Provider, error) {
 		return nil, err
 	}
 
-	fmt.Sprintf("Executor type: %s, Executor config: %v", sc.ExecutorType, sc.ExecutorConfig)
 	logger.Info("Creating Spark store with type:", sc.StoreType)
-	serializedDatabricksConfig, err := sc.StoreConfig.Serialize()
+	serializedFilestoreConfig := sc.StoreConfig.Serialize()
```
Comment (sdreyer) [inline provider/spark.go:486]: Serialize() should return an error instead of panic

File: provider/spark.go [hunk 6]
```diff
@@ -516,28 +542,32 @@ func (e EMRExecutor) InitializeExecutor(store FileStore) error {
...
-	// 		client:      client,
-	// 		logger:      logger,
-	// 		clusterName: emrConfig.ClusterName,
-	// 	}
-	// 	return &emrExecutor, nil
-	// } else
-	if execType == Databricks {
-		return NewDatabricksExecutor(config)
+func NewSparkExecutor(execType SparkExecutorType, config SparkExecutorConfig, logger *zap.SugaredLogger) (SparkExecutor, error) {
+	if execType == EMR {
```
Comment (sdreyer) [inline provider/spark.go:546]: These should be the same level of abstraction. Should have a NewEMRExecutor function that does this

File: provider/spark.go (hunk 3 above)
Comment (ahmadnazeri) [reply provider/spark.go:78]: FileStoreConfig already exists; it's of type `[]byte`

**Pull request merged**